sections/shared/Layout/AppLayout/Header/WalletActions.tsx

```diff
@@ -100,7 +100,7 @@ export const WalletActions: FC = () => {
 
 	useEffect(() => {
 		if (signer) {
-			setWalletLabel('loading...');
+			setWalletLabel(truncatedWalletAddress!);
 			signer.getAddress().then((account: string) => {
 				const _account = account;
 				getENSName(_account, staticMainnetProvider).then((_ensName: string) => {
```

Comment:
how do we feel about something like this?

```
useEffect(() => {
	if (signer) {
		setWalletLabel('');
		signer
			.getAddress()
			.then((account: string) => {
				const _account = account;
				getENSName(_account, staticMainnetProvider).then((_ensName: string) => {
					setEns(_ensName);
					setWalletLabel(_ensName || truncatedWalletAddress!);
				});
			})
			.catch((_) => setWalletLabel(truncatedWalletAddress!));
	}
}, [signer, truncatedWalletAddress]);
```

Reply:
yeah that makes sense - ok sounds good!

Reply:
AAVE is a good example of this ^^